### PR TITLE
Content for Create an API token

### PR DESCRIPTION
**Pull request by turnipdabeets**
https://beyondidentity.atlassian.net/browse/DEVX-1173

- Explain where to get an API Token
- Explain how to get a token visually from console
- Explain Token scopes and expiration (Link to our detailed Scopes page)
- Explain how to get a token via client_id / client_secret (later)

### CODE REVIEW

**Review by sublimezeal — COMMENTED** [review]
This is my first review so please Slack me if these comments don't make sense! I offered a few comments to clarify and make the examples more specific.

File: docs/workflows/api-token.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -d "grant_type=client_credentials&scope=$SCOPES"
+```
+
+ </TabItem>
+ <TabItem value="node" label="Node">
+
+```jsx
+const apiTokenResponse = await fetch(
+  `https://api-${REGION}.beyondidentity.com/v1/tenants/${TENANT_ID}/realms/${REALM_ID}/applications/${APPLICATION_ID}/token`,
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md]: Good idea calling out - added that! 

As far as not working in postman - I tested the curl in postman before posting the code snippet and the node example is copied from a production app, but I'll test the curl again to make sure.

File: docs/workflows/api-token.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    body: (() => {
+      let formData = new URLSearchParams();
+      formData.append('grant_type', 'client_credentials');
+      return formData;
+    })(),
+    headers: {
+      Authorization:
+        'Basic ' +
+        Buffer.from(
+          `${MANAGEMENT_API_CLIENT_ID}:${MANAGEMENT_API_CLIENT_SECRET}`
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md:74]: added!

File: docs/workflows/api-token.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+ toc_min_heading_level: 2
+ toc_max_heading_level: 2
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Create an API Token
+
+All [Beyond Identity API](https://developer.beyondidentity.com/api/v1) endpoints require authentication using an access token. The access token is generated through OAuth 2.0 or OIDC, using the authorization code flow or the client credentials flow.
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md:14]: I added a section for this and also for revoking tokens via API

File: docs/workflows/api-token.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+You can obtain an access token either through:
+
+1. [Beyond Identity Admin Console](api-token#create-token-with-the-beyond-identity-admin-console)
+2. [Beyond Identity APIs](api-token#create-token-with-the-beyond-identity-api)
+
+The access token will work for any application that you have configured to provide access to the Beyond Identity Management API Resource Server. The "Beyond Identity Management API" application is provided by default as part of the tenant onboarding process.
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md:21]: I'm still trying to get clarity on this but I think this is a key part, your application has to be connected to the resource server for the access token to work.

[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md]: updated!

File: docs/workflows/api-token.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (sublimezeal) [reply docs/workflows/api-token.md:14]: i like it!

**Comment by ismailmustafa** [comment]
A bit of a nit, but could we use https://curlconverter.com/ to generate code examples for [node, go, python] to start with? It would be an easy way to make it more approachable to a wider variety of developers.

**Review by Will5 — APPROVED** [review]

File: docs/workflows/api-token.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+  log.Fatal(err)
+}
+fmt.Printf("%s\n", bodyText)
+}
+```
+
+</TabItem>
+</Tabs>
+
+2. Create an access token the with authorization code:
```
Comment (tomsik68) [inline docs/workflows/api-token.md:215]: I think we should somehow clarify that creating an access token happens in the callback - the thing that serves redirect_uri. I'm not sure how exactly we might want to write that, but I see some options:

1. Have a separate section about callback and what needs to be done there (extract code from request, check state etc.)
2. Start this section by saying that this needs to happen in callback where we receive the grant code

File: docs/workflows/api-token.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+</TabItem>
+</Tabs>
+
+### Revoking Access Tokens:
+
+In order to revoke an access token, you must be passed authentication in the form of either Bearer or Basic.
+
+In the case of Basic authentication, the passed authentication must come from an Application registered in auth-config-service. The passed access token must be signed by the same client id as the application.
+
+In the case of Bearer authentication, the passed authentication must contain the scope "token:delete".
```
Comment (tomsik68) [inline docs/workflows/api-token.md:310]: The scope is `tokens:delete`. We use plural in most places.

File: docs/workflows/api-token.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+### Revoking Access Tokens:
+
+In order to revoke an access token, you must be passed authentication in the form of either Bearer or Basic.
+
+In the case of Basic authentication, the passed authentication must come from an Application registered in auth-config-service. The passed access token must be signed by the same client id as the application.
```
Comment (tomsik68) [inline docs/workflows/api-token.md:308]: I'd suggest rephrasing this:

>  In the case of Basic authentication, the passed authentication must come from the confidential Application that the token has been minted for.

I don't think it's useful to say that the application is registered in auth-config-service.

File: docs/workflows/api-token.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+
+In the case of Bearer authentication, the passed authentication must contain the scope "token:delete".
+
+Note that passing an invalid token, or a token which has already been revoked or expired, will produce a success response, pursuant to [RFC7009§2.2](https://www.rfc-editor.org/rfc/rfc7009).
+
+<Tabs groupId="api-token-platform" queryString>
+<TabItem value="curl" label="Curl">
+
+```bash title="/revoke"
+curl https://auth-$REGION.beyondidentity.com/v1/tenants/$TENANT_ID/realms/$REALM_ID/applications/$MANAGEMENT_APPLICATION_ID/revoke \
```
Comment (tomsik68) [inline docs/workflows/api-token.md:318]: I think we're missing authorization for this request.

File: docs/workflows/api-token.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+The access token should be generated with your Beyond Identity Management API application in order to access any of the Beyond Identity APIs. The "Beyond Identity Management API" application is provided by default as part of the tenant onboarding process.
+
+The access token must be provided in the Authorization header of an API request like the following:
+
+```bash
+curl https://api-us.beyondidentity.com/v1/... \
+-X $HTTP_METHOD -H "Authorization: Bearer $TOKEN"
+```
+
+API access tokens are valid for 3 months (TTL 7776000 seconds). You can restrict the token's access with [scopes](../apis/scopes) by selecting a list in the Beyond Identity Admin Console or specifying a space-separated string of scopes in your API request.
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md:30]: I didn't see that in the API spec. How would you send that? Like this?

```
curl https://auth-$REGION.beyondidentity.com/v1/tenants/$TENANT_ID/realms/$REALM_ID/applications/$MANAGEMENT_APPLICATION_ID/token \
-X POST \
-u "$MANAGEMENT_API_CLIENT_ID:$MANAGEMENT_API_CLIENT_SECRET" --basic \
-H "Content-Type: application/x-www-form-urlencoded" \
-d "grant_type=client_credentials&scope=$SCOPES&expires=12345"
```

File: docs/workflows/api-token.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+```bash title="/authorize"
+curl https://auth-$REGION.beyondidentity.com/v1/tenants/$TENANT_ID/realms/$REALM_ID/applications/$MANAGEMENT_APPLICATION_ID/authorize?response_type=code&client_id=$MANAGEMENT_API_CLIENT_ID&redirect_uri=$REDIRECT_URI&scope=openid&state=$STATE&code_challenge=$OPTIONAL_CODE_CHALLENGE&code_challenge_method=S256
+```
+
+</TabItem>
+<TabItem value="node" label="Node">
+
+```jsx title="/authorize"
+fetch(
+  'https://auth-REGION.beyondidentity.com/v1/tenants/TENANT_ID/realms/REALM_ID/applications/MANAGEMENT_APPLICATION_ID/authorize?response_type=code'
```
[the remark this answers is not in the document]
Comment (turnipdabeets) [reply docs/workflows/api-token.md:160]: thanks for catching this! I was using a code converter

File: docs/workflows/api-token.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (tomsik68) [reply docs/workflows/api-token.md:30]: We currently don't support the request parameter that you described. All the tokens for an application have the same TTL so you can't adjust it for every token separately. You can, however, change the value by using PATCH on an Application or changing that in console.

File: docs/workflows/api-token.md (hunk 7 above)
Comment (tomsik68) [reply docs/workflows/api-token.md:308]: I agree, let's update the source specs too to prevent confusion.

**Pull request merged**